tree-sitter/src/TreeSitter/GenerateSyntax.hs

```diff
@@ -128,8 +111,8 @@ ctorForTypes named constructorName types = recC (toName named constructorName) r
 
 
 -- | Convert field types to Q types
-fieldTypesToNestedEither :: NonEmpty TreeSitter.Deserialize.Type -> Name -> Q TH.Type
-fieldTypesToNestedEither xs typeParameterName = foldr1 combine (fmap convertToQType xs) `appT` varT typeParameterName
+fieldTypesToNestedSum :: NonEmpty TreeSitter.Deserialize.Type -> Q TH.Type
+fieldTypesToNestedSum xs = foldr1 combine (fmap convertToQType xs)
```

Comment:
Prior to changing the type signature/what args this function accepts, I wasn't understanding the error message I was seeing until @robrix clarified the behavior was related to type synonyms, so writing down the reasoning here. 

Suppose you have the following:
```
type Foo = (Maybe :+: Maybe) a
type Bar a = (Foo :+: Maybe) a
```

This won't compile, because type synonyms can't be partially applied.

By contrast, this compiles:
```
type Foo = (Maybe :+: Maybe)

type Bar a = (Foo :+: Maybe) a
```

In the case we need to construct a type with a `typeParameterName`, as we do in the call to `fieldTypesToNestedSums` within `ctorForProductType`, I just use `appT` to create that there.